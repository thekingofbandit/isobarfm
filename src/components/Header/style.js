import styled from 'styled-components';

export const Container = styled.div`
  background-color: #333;
  display: flex;
  align-items: center;
`;
