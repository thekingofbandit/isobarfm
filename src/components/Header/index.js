import React from 'react';
import { Link } from 'react-router-dom';
import Logo from '../Logo';
import { Container } from './style';

const link = 'http://pngimg.com/uploads/apple_logo/apple_logo_PNG19670.png';

const Header = ({ routes }) => (
  <Container>
    <Logo alt="logo" src={link} />
    {routes.map(({ displayName, to }) => (
      <Link {...{ to, key: to }}>{displayName}</Link>
    ))}
  </Container>
);

Header.defaultProps = { routes: [] };

export default Header;
