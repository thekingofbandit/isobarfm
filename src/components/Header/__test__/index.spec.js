import React from 'react';
import { shallow } from 'enzyme';
import Header from '../';

describe('Header', () => {
  it('should render header', () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });

  it('should render header with routes', () => {
    const props = {
      routes: [
        { to: 'route1', displayName: 'route1' },
        { to: 'route2', displayName: 'route2' },
        { to: 'route3', displayName: 'route3' },
        { to: 'route4', displayName: 'route4' },
      ],
    };
    const component = shallow(<Header {...props} />);
    expect(component).toMatchSnapshot();
  });
});
