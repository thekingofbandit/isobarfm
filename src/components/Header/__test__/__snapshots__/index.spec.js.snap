// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render header 1`] = `
<styled.div>
  <Logo
    alt="logo"
    src="http://pngimg.com/uploads/apple_logo/apple_logo_PNG19670.png"
  />
</styled.div>
`;

exports[`Header should render header with routes 1`] = `
<styled.div>
  <Logo
    alt="logo"
    src="http://pngimg.com/uploads/apple_logo/apple_logo_PNG19670.png"
  />
  <Link
    key="route1"
    to="route1"
  >
    route1
  </Link>
  <Link
    key="route2"
    to="route2"
  >
    route2
  </Link>
  <Link
    key="route3"
    to="route3"
  >
    route3
  </Link>
  <Link
    key="route4"
    to="route4"
  >
    route4
  </Link>
</styled.div>
`;
